### ISSUE TITLE
Invalid translation of GRAPH and nested queries

### ISSUE THREAD

**Issue by rubensworks** [joachimvh/SPARQLAlgebra.js#50]
When this query is given as input:
```sparql
select ?x where {
graph ?g {
  {select ?x where {?x ?p ?g}}
}
}
```

The following algebra is produced (in quad-mode):
```json
{
  "type": "project",
  "input": {
    "type": "project",
    "input": {
      "type": "bgp",
      "patterns": [
        {
          "subject": {
            "value": "x"
          },
          "predicate": {
            "value": "p"
          },
          "object": {
            "value": "g"
          },
          "graph": {
            "value": "g"
          },
          "type": "pattern"
        }
      ]
    },
    "variables": [
      {
        "value": "x"
      }
    ]
  },
  "variables": [
    {
      "value": "x"
    }
  ]
}
```

This is however incorrect, as the `?g` variable from the `graph` clause is not the same as the `?g` variable within the inner `select` clause, as the latter one is projected away.

One solution may be to detect these cases, and use a non-clashing dummy variable name for the inner query.

Discovered in https://github.com/comunica/comunica/issues/287 (SPARQL 1.1 test http://www.w3.org/2009/sparql/docs/tests/data-sparql11/subquery/manifest#subquery03)

**Comment by joachimvh** [comment]
Fixed in 31f569a

**Comment by rubensworks** [comment]
@joachimvh Could you bump the dependency version in Comunica, and check off the box for the corresponding test that failed because of this in https://github.com/comunica/comunica/issues/287 ?
